numba/typed/typedobjectutils.py

```diff
@@ -197,3 +197,78 @@ def build_wrapper(fn):
     builder.ret(context.get_constant(types.int32, -1))
 
     return equal_fn
+
+
+def _get_primitive_equal(context, module, datamodel, element_type):
+    assert not datamodel.contains_nrt_meminfo()
+
+    data_ptr_ty = ir.IntType(8).as_pointer()
+    equal_fnty = ir.FunctionType(ir.IntType(32), [data_ptr_ty, data_ptr_ty])
+    equal_fn = cgutils.get_or_insert_function(
+        module,
+        equal_fnty,
+        name='.numba_{}.{}_equal'.format(
+            context.fndesc.mangled_name, element_type),
```

Comment:
NIT: you can use an f-string here

Reply:
Sure, just the legacy code used this format, I just followed it..
Can change to f-format.